### PR TITLE
Migrant Job DLC - Strike the Earth!

### PR DESCRIPTION
**Pull request by timothymtorres**
## About The Pull Request
migrants_arriving

This adds a new job called "Migrant" that forces a player to spawn at a lavaland or ice outpost that is located randomly in the wasteland.  The challenge is if you can make it to the station alive!

<details>
<summary>Spoilers:</summary>

dreamseeker_2JJOX95wgT

dreamseeker_H7QXPwRlHA

</details>

### TODO

- [ ] Add more ID access so migrants can enter the station mining bases
- [ ] Add adminbus option to force players to spawn at migrant outpost
- [ ] Add some survival gear to players spawn with the quirk (survival knife, extended oxygen tank, etc.)

## Why It's Good For The Game
- More use for landing field designator and equipment
- Harder difficulty for people who want a challenge
- More interactions for miners and off-station ghost roles who come across migrants

The bottom line is it's meant to be optional.

## Changelog
:cl:
add: Add new migrant shelter ruin that is always guaranteed to spawn on lavaland or ice.  Comes equip with some basic survival gear.
add: Add new migrant job that forces a person to spawn off the station inside the migrant shelter by default.  They must try to reach the station and survive the wilderness. Strike the earth!
/:cl:

### CODE REVIEW

**Comment by private-tristan** [comment]
> spawn in as migrant
~~bronze~~ colossus outside
mfw

**Comment by zxaber** [comment]
It's really odd that they'd already have an NT-issued ID and other job gear. If you want it otherwise to be a standard job (rather than a ghost role that needs to get a job from the HoP), you should theme it as "Crashed Arrival Pod". You could probably even have them arrive to a random spot in the wastes in a delivery pod as they spawn.

**Comment by Stalkeros2** [comment]
Finally, Lava Dwarves 2

**Comment by flowercuco** [comment]
i know its a different thing but i feel like it would be better to like, do changes to the existing lavaland denizen ghost spawner stuff than have someone spawn not in the part of the game where all of the stuff for their job is? an engineer has tools that can be used for some makeshift stuff if he digs iron out randomly from walls that they dont know is there bc they dont have a mining analyzer. 

imagine if someone picked this quirk while being the captain its like theres no acting captain bc the captain spawns with the safe code and so if theres no HoP no one can mess with anyones access for like half an hour assuming they even make it back to the station.

**Comment by lizardqueenlexi** [comment]
I like the idea, but... I don't think it should be a quirk. It feels somewhat likely to cause problems if people are spawning offstation and quite likely dying on the way back... while taking up a specific job slot that might be needed _on_ the station. Plus, there's nothing to limit how _many_ people can have the same quirk, which could lead to a chunk of the station just not being there.

It'd probably be more suitable as a ghost spawner or as a custom job like Prisoner - to limit the impact on the crew and the number of people doing it at the same time.

**Comment by zxaber** [comment]
>There is nothing to stop players from picking jobs and then running off to explore space. 

For the record, we do have administrative restrictions on doing this for heads of staff. So keeping this from affecting heads may be a good idea.

This doesn't affect AIs, does it? Quirks don't generally affect silicons, but I'm not sure if that's just because quirks tend to be organic-focused.

**Comment by datorangebottle** [comment]
I feel like they should get proper lavaland weapons instead of just survival knives. Those things are practically worthless. Crushers, maybe, since those are bulky and aren't particularly useful outside of the mining areas.

**Comment by Wallemations** [comment]
If you really wanted to make it roundstart and make sure not too many people spawn there at the start, why not make it a roundstart job like prisoner? It being a quirk doesn't make much sense to me personally.

It'd probably most fit as a ghost spawner if anything tbh

**Comment by thgvr** [comment]
This should be a ghost spawner if anything, not a quirk that rips people off of the station.

**Comment by rellyreri** [comment]
what happens if 2+ people take the quirk?

**Comment by JohnFulpWillard** [comment]
I dunno how I feel about it being a quirk anyone can take.
I would like to put forward the idea of a station trait that moves a random job (or prisoners) there, making it less optional as it would lead to funnier situations

**Comment by Sealed101** [comment]
looks fine to me as a personal quirk, adds the optional challenge as well as spices up hardcore random gameplay

**Comment by newfren** [comment]
this should be added because it is very funny

**Comment by DemoCrack** [comment]
Not enough dorf.

**Comment by rasepretep** [comment]
... was someone watching me when I did this exact thing a while ago, with the help of an admin teleporting me to a far-off side of lavaland roundstart?

The idea was I was on an escape pod stranded on Lavaland from previous shift and had to make my way back to the station. Nice as that migrant base is, It would be a lot better if the person spawned there with minimal resources or shelter. The migrant's goal is to make it to the station, not live on lavaland, it's not a hermit ghost role.

This all comes from me playing Ashssistant enough times to be able to survive on Lavaland fairly reliably, so lemme add input from those experiences.

I propose spawning like so:
- In a crashed escape pod, you're not a migrant, you're crew from previous shift
- You have the usual roundstart gear you spawn with including your headset. You can always just call for help or you can toss the headset to give yourself more challenge.
- Escape pod locker contains: a pickaxe, brute medkit, 2 emergency air tanks, mining winter coat, combat knife (normally I would say you don't need the combat knife, because you can dig sand, make a glass shard, kill a couple lobsters to get a bone dagger, but it's unnecessary to force the challenge)
- No GPS. You have to figure out which way to go, plus it stops miners from beelining to you. It's a question whether that could be optional and the locker contains a GPS that's turned off and you are given the choice to pick it up and turn it on if you wish to make it easier on yourself.
- Your goal is to make it to the station, not live on Lavaland, so you shouldn't have unlimited supplies. It's a challenge, not a hermit spawn.

My experiences as Ashssistant:
- Your main problem is not having enough medicine. You have to make do with legion cores, cactus fruit and mushroom leaves once you run out of any meds you had on you. It's not as hard as you might think, if you know what you're doing.
- Your main danger is moving too fast and running into enemies. You have to move slow, tunnel when it's safer and have a place to withdraw to when you anger too many mobs.
- Having an inexhaustible way to heal yourself, like a medibot or good virus, makes the whole challenge a breeze.
- Shelter is never a problem as long as you have a pickaxe. 2 ash tiles get you 10 sand to make sandstone and a door from that. seal youself in a small nook you dug out and designate the area, you have shelter from the storm now.
- Very often you get lucky and stumble upon a ruin that makes it a lot easier. Graveyard that has tons of bones and a MODsuit, the derelict pod with a dead miner and so on.
- Once you manage to make a bone axe, everything becomes a bit easier, although goliaths will always be extremely dangerous to melee.
- Any miner legion you kill has an extended capacity emergency air tank. You're very unlikely to run out of air before getting to the station.
- Sooner or later miners will find you or you find them.

In summary, I love this idea and I have been basically playing this extensively whenever I rolled Ashssistant and went to live on lavaland with minimal resources. It's a lot easier than it seems, but it does require knowing how to survive there.

**Review by Mothblocks — CHANGES_REQUESTED** [review]
This should be a station trait that applies to prisoners. The only issue there is the wiki tells new players to pick prisoner somewhere (and that has always been stupid)

**Comment by Iamgoofball** [comment]
@Mothblocks that's a completely different mechanic entirely and goes against the PR author's design

@timothymtorres i advise you as a bystander to appeal to the other headcoders

**Comment by Mothblocks** [comment]
yes its a completely different mechanic because the original idea of a quirk that applies to any job is not something i want in the game, it is called a compromise

**Comment by rasepretep** [comment]
Make it 2 "Marooned Crew" job slots (like Prisoners have), spawn in an escape pod on Lavaland as Assistants with assistant IDs, only allowed roundstart, there's your compromise. Forcing prisoners into it via a station trait seems like a bad idea, it should be for people who want to try it and are confident they can survive, not for people to be thrown into it.

**Comment by Iamgoofball** [comment]
@Mothblocks if you want a different feature then was given to you for free why don't you write it yourself or offer @timothymtorres a fair wage to write it

**Comment by timothymtorres** [comment]
I've decided to make this into it's own job since a lot of people have concerns of this affecting on-station job slots.  Not a fan of making this a station trait for prisoners or making this a ghost role especially since we already have hermit.

**Comment by newfren** [comment]
this became much less funny and now just sorta steps on miners' toes :(

**Comment by Fikou** [comment]
not a fan of a new job thats just assistant but also miner. the role of a migrant does not work when its just a guy that has a weirder spawn. (and it makes little sense that they would have a special place only to travel to lavaland and then to station, why wouldnt they just take that shuttle to the station?) this job being antag would also be troublesome as they spawn on lavaland with no other crew and can just waste their threat by dying to some goliath. but if its not antag, you can just trust them (and like, they just go through the cargo shuttle and become a crewmember like normal, thats weird)
all of them having an issued id with the name migrant also feels very ooc and gamey
other *cough* servers *cough* and dwarf fortress itself makes migrants much more special, with a whole group of migrants, and each migrant serving a role that holds some meaning to the fort (being masters in various skills/having unique gear).
i think mothblocks suggestion of station trait for prisoners is your best choice here, if you do it (and its done and not wip) tell me and i can reopen the pr (or make a new one yourself)

**Comment by Fikou** [comment]
also a random event like fugitives that spawns a random group of ghosts as random picked custom jobs could be cool but then you run into problems with metagaming

**Comment by timothymtorres** [comment]
RIP 🪦

**Comment by Farquaar** [comment]
No human is illegal smh

**Comment by bruhman8ghughug** [comment]
migrant??? like lifebebrbebeb?
